### PR TITLE
Add java 17 preview image

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@timja would it make sense to start a GitHub project for Java 17 related changes so that we could track the required changes?

**Comment by timja** [comment]
> @timja would it make sense to start a GitHub project for Java 17 related changes so that we could track the required changes?

sure

**Comment by timja** [comment]
https://github.com/orgs/jenkinsci/projects/6

**Comment by timja** [comment]
Windows is also failing in the master branch...

**Review by MarkEWaite — APPROVED** [review]
Looks good to me.  I confirmed that the JENKINS_ENABLE_FUTURE_JAVA is no longer required with Jenkins versions using the recent executable war file.  It recognizes Java binary format 55 (Java 11) as a supported version

File: jenkins.sh [hunk 1]
```diff
@@ -34,7 +34,18 @@ if [[ $# -lt 1 ]] || [[ "$1" == "--"* ]]; then
     jenkins_opts_array+=( "$item" )
   done < <([[ $JENKINS_OPTS ]] && xargs printf '%s\0' <<<"$JENKINS_OPTS")
 
-  exec java -Duser.home="$JENKINS_HOME" "${java_opts_array[@]}" -jar ${JENKINS_WAR} "${jenkins_opts_array[@]}" "$@"
+  FUTURE_OPTS=""
+  if [[ "$JENKINS_ENABLE_FUTURE_JAVA" ]] ; then
+    FUTURE_OPTS="--add-opens java.base/java.lang=ALL-UNNAMED
```
Comment (timja) [inline jenkins.sh:39]: another option is we could just add these for Java 11 onwards, it gets rid of the illegal reflective access warnings

**Comment by timja** [comment]
> Windows is also failing in the master branch...

fixed in https://github.com/jenkinsci/docker/pull/1213

File: jenkins.sh (hunk 1 above)
Comment (MarkEWaite) [reply jenkins.sh:39]: I like the continuing warnings for illegal reflective access on Java 11 myself.  I hope that we'll continue to make progress removing those warnings.

Comment (timja) [reply jenkins.sh:39]: some will get fixed by Guice upgrade, the rest are more difficult, XStream usage needs to be investigated to see if we can avoid using the default converters and jboss marshalling needs us to compile with at least java 11 IIRC

**Pull request merged**